format-avro/src/main/java/io/cdap/plugin/format/avro/input/AvroInputFormatProvider.java

```diff
@@ -117,32 +118,40 @@ public Schema getDefaultSchema(@Nullable FormatContext context) throws IOExcepti
     String filePath = conf.getProperties().getProperties().getOrDefault("path", null);
     SeekableInput seekableInput = null;
     FileReader<GenericRecord> dataFileReader = null;
+    GenericRecord firstRecord = null;
     try {
       Job job = JobUtils.createInstance();
       Configuration hconf = job.getConfiguration();
       // set entries here, before FileSystem is used
       for (Map.Entry<String, String> entry : conf.getFileSystemProperties().entrySet()) {
         hconf.set(entry.getKey(), entry.getValue());
       }
-      Path file = conf.getFilePathForSchemaGeneration(filePath, ".+\\.avro", hconf, job);
-      DatumReader<GenericRecord> dataReader = new GenericDatumReader<>();
-      seekableInput = new FsInput(file, hconf);
-      dataFileReader = DataFileReader.openReader(seekableInput, dataReader);
-      GenericRecord firstRecord;
-      if (!dataFileReader.hasNext()) {
-        return null;
+      List<Path> paths = conf.getFilePathsForSchemaGeneration(filePath, ".+\\.avro$", hconf, job); 
+      for (Path file : paths) {
+        DatumReader<GenericRecord> dataReader = new GenericDatumReader<>();
+        seekableInput = new FsInput(file, hconf);
+        try {
+          dataFileReader = DataFileReader.openReader(seekableInput, dataReader);
+          firstRecord = dataFileReader.next();
+          return new AvroToStructuredTransformer().convertSchema(firstRecord.getSchema());
+        } catch (Exception e) {
```

Comment:
Nit: We are capturing all kinds of exception here. We should log the exception to make it easy to debug (e.g. to separate issues such as are we not loading the file because we don't have read permissions? or because it's invalid?)

Reply:
This could lead to noisy logging, but I think logging exceptions here is a good idea. In general, I think customers should be aware if some of their avro files might be formatted incorrectly, or if there are permission issues for some files in the directory.

Reply:
OK. So if we find an invalid file (no permission or something cause fail to read), the loop will stop and an error will be shown in the UI. And if we find an empty file, we will just skip it and check the next file.  Is that correct?